docs/config/postprocess_docs.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+import os
+from bs4 import BeautifulSoup
+import pybtex.database
+import re
+
+def append_eprint_links_to_citelist(html_dir, references_file):
+    logger = logging.getLogger(__name__)
+
+    # Load `citelist.html` file as structured HTML
+    with open(os.path.join(html_dir, 'citelist.html'), 'r') as citelist_file:
+        citelist = BeautifulSoup(citelist_file, 'html.parser')
+    # Load `References.bib` as structured bibliography database
+    references = pybtex.database.parse_file(references_file)
+
+    for citeref_anchor in citelist.find_all(id=re.compile('CITEREF_*')):
+        key = re.match('CITEREF_(.*)', citeref_anchor.attrs['id']).groups()[0]
+        entry = references.entries[key]
+        # We support only arXiv eprints for now. More formats may be added here
+        # if necessary.
+        if 'archivePrefix' in entry.fields \
+            and entry.fields['archivePrefix'] == 'arXiv' \
+            and 'eprint' in entry.fields:
+            logger.debug("Found arXiv link for {}.".format(key))
+            arxiv_url = "https://arxiv.org/abs/{}".format(
+                entry.fields['eprint'])
+            eprint_link = citelist.new_tag('a', href=arxiv_url)
+            eprint_link.string = "arXiv:{}".format(entry.fields['eprint'])
+        else:
+            logger.debug("Found no supported eprint data for {}.".format(key))
+            continue
+        # Append the eprint link to the citation HTML
+        citeref_item = citeref_anchor.find_next('p')
+        citeref_item.append(eprint_link)
+        citeref_item.append('.')
+        logger.info("Added eprint link to {}: {}".format(key, eprint_link))
+
+    # Write the `citelist.html` file back to disk
+    with open(os.path.join(html_dir, 'citelist.html'), 'w') as citelist_file:
+        citelist_file.write(str(citelist))
+
+if __name__ == '__main__':
+    import argparse as ap
```

Comment:
Could you please separate the argument parsing into a separate `parse_args` function?

Reply:
why do you prefer that? the argument parsing is only relevant when called as the `__main__` module. Could be cleaner this way, what do you think?

Reply:
I think it's less clean this way because you have a fairly long block of code doing two completely unrelated things. It's better to factor code into functions that do one single thing, e.g. `parse_args`. If you wrote it as a separate function is very easy to look at and say "Ah, that just parses the args, I'll focus on understanding the rest right now", whereas the way you wrote it now variables have a much larger scope than they need to.

Reply:
makes sense, thanks